custom_components/rct_power/sensor.py

```diff
@@ -35,6 +37,15 @@ async def async_setup_entry(
         for entity_description in battery_sensor_entity_descriptions
     ]
 
+    battery_binary_sensor_entities = [
```

Comment:
If I understand HA's component architecture correctly, setting up a binary sensor in `sensor.py` would not be correct. Instead, it should be set up as part of the `binary_sensor` platform (meaning in `binary_sensor.py`), which is being delegated to by the `__init__.py`'s loop over the `PLATFORMS` list (which would also have to be extended to include binary sensors). In light of that I actually like your alternative suggestion of introducing a single sensor that displays the state using an enum. What do you think?

Reply:
TBH, I'm not fully aware of the component architecture, as I never built an integration myself. But it makes sense, even though it just seems to work in my installation without complying with the convention.
But nevertheless, I also think we should go for the enum alternative. I just thought about this after I made the current implementation.
I will refactor this and update the PR once I tested it.